src/name/ip_address.rs

```diff
@@ -295,10 +295,36 @@ pub(super) fn presented_id_matches_constraint(
     let mut name = untrusted::Reader::new(name);
     let mut constraint_address = untrusted::Reader::new(constraint_address);
     let mut constraint_mask = untrusted::Reader::new(constraint_mask);
+    let mut seen_zero_bit = false;
+
     loop {
         let name_byte = name.read_byte().unwrap();
         let constraint_address_byte = constraint_address.read_byte().unwrap();
         let constraint_mask_byte = constraint_mask.read_byte().unwrap();
+
```

Comment:
Maybe also add a bit of commentary at the top of the loop that this loop runs on a byte-per-iteration basis through `name`, `constraint_address` and `constraint_mask`?